Mage/src/main/java/mage/game/permanent/token/ElementalToken.java

```diff
@@ -58,5 +59,18 @@ public ElementalToken(String setCode, int tokenType) {
 
         this.setOriginalExpansionSetCode("CON");
     }
-
+    
+    public ElementalToken(String setCode, int tokenType, boolean hasHaste) {
```

Comment:
can also refactor all existing instances of Elemental Token to use this class instead to reduce some code duplication throughout.